### PR TITLE
Fix DefaultAzureCredential exception issues #8936

### PR DESCRIPTION
**Pull request by tzhanl**
ManagedIdentityCredential is unavailable No managed identity endpoint found..

- It is missing a period after unavailable
- It ends with two periods

SharedTokenCacheCredential is unavailable No accounts were found in thecache. To authenticate with the SharedTokenCacheCredential, login an account through developer tooling supporting Azure single sign on..

- Same as above. Missing period. Ends with two periods.
- "thecache" needs a space.
- "login" should be "log in"
- Suggest this: "To authenticate with the SharedTokenCacheCredential, log in with a development tool that supports Azure single sign on."

**Solution of period problem**:
While creating Aggregate Exception message, don't take care of period. Let inner message take care of period itself.

### CODE REVIEW

**Review by jongio — COMMENTED** [review]
Please compare this to the message that is outputted by the other SDKs. Python, Java, JS. 

I believe Python has the aggregate handle the period. And let’s get consistent verbiage for them all.  See this PR where I removed a period from inner for Py https://github.com/Azure/azure-sdk-for-python/pull/8594

I understand it will be separate PRs for the other languages and updates to this one.

File: sdk/identity/Azure.Identity/src/SharedTokenCacheCredential.cs [hunk 1]
```diff
@@ -19,9 +19,9 @@ namespace Azure.Identity
...
         internal const string MultipleAccountsInCacheMessage = "Multiple accounts were found in the cache. To authenticate with the SharedTokenCacheCredential, set the AZURE_USERNAME and AZURE_TENANT_ID environment variables to the preferred username and tenantId, or specify them to the constructor. {0}";
-        internal const string NoMatchingAccountsInCacheMessage = "No account matching the specified{0}{1} was found in the cache. To authenticate with the SharedTokenCacheCredential, login an account through developer tooling supporting Azure single sign on. {2}";
+        internal const string NoMatchingAccountsInCacheMessage = "No account matching the specified{0}{1} was found in the cache. To authenticate with the SharedTokenCacheCredential, login an account with a development tool that supports Azure single sign on. {2}";
         internal const string MultipleMatchingAccountsInCacheMessage = "Multiple accounts matching the specified{0}{1} were found in the cache. To authenticate with the SharedTokenCacheCredential, set the AZURE_USERNAME and AZURE_TENANT_ID environment variables to the preferred username and tenantId, or specify them to the constructor. {2}";
```
Comment (schaabs) [inline sdk/identity/Azure.Identity/src/SharedTokenCacheCredential.cs]: Should these all use the same terminology. The first uses Local account whereas the others use something like "Account in the cache". Should we just remove the reference to the cache from all of them and just refer to local accounts?

Comment (tzhanl) [reply sdk/identity/Azure.Identity/src/SharedTokenCacheCredential.cs]: remove "in the cache", add "local" into other messages.

**Comment by schaabs** [comment]
@tzhanl thanks for your work on this. I have merged these changes as part of PR #9821

**Pull request closed without merging**